src/main.cpp

```diff
@@ -856,6 +856,9 @@ int main ( int argc, char** argv )
                 // only start application without using the GUI
                 qInfo() << qUtf8Printable( GetVersionAndNameStr ( false ) );
 
+                // enable server list if a directory server is defined
```

Comment:
So the title says it's about GUI mode.  This change is explicitly non-GUI, even in the comment...

Reply:
The problem was evident in Gui mode, but the fix required the non-Gui item you flagged so that headless still worked.

The problem was that every time the central server input field lost focus and fired the “editing finished” event, the function to update the central server was called. The logic in that function was setting `bEnabled` irrespective of the checkbox, presumably a hangover from when there was no such checkbox.

Now, `bEnabled`is only ever controlled from:
1. The checkbox,
2. Reading the ini file,
3. The headless initialisation (now needed because it had been removed from the above function).

Reply:
Try the procedure listed in #1624 on 3.7.0 and you’ll see what I mean.

Reply:
Anyway, I don't like comments that repeat code.  Say "why", not "what"...